modules/admin-ui/src/main/webapp/scripts/shared/services/resourceHelperService.js

```diff
@@ -44,28 +44,25 @@ angular.module('adminNg.services')
 
             var effectiveMapper = mapper || defaultMapper;
 
-            var result = [], data = {};
+            var result = [],
+                data = {};
 
-            try {
-              data = JSON.parse(responseBody);
-
-              // Distinguish between single and multiple values
-              if ($.isArray(data.results)) {
-                angular.forEach(data.results, function (event) {
-                  result.push(effectiveMapper(event));
-                });
-              } else {
-                result.push(effectiveMapper(data.results));
-              }
-            } catch (e) {
-              //console.warn(e);
-            }
+            data = JSON.parse(responseBody);
 
             // guard against empty dataset
-            if (angular.isUndefined(data.results)) {
+            if (data === null || angular.isUndefined(data.results)) {
```

Comment:
@lkiesow As this behaviour seems weird to me, I digged somewhat into what is causing JSON.parse to result in _null_. The cause is that responseBody is already _null_ and _JSON.parse(null)_ results in _null_.
This happens, for example, if a request is cancelled (or when ever the server responds with an empty body).

Maybe it would be better to check _(responseBody === null_) to make the code easier to understand.

What do you think?

Reply:
That would only catch half of the cases: If the returned JSON would be null (Important: JSON=null not JS-Value=null), you would still run into the same error. I can add a second check before parsing the response if you want me to. Though that's then more or less redundant.

Though in any case, `angular.isObject` may be a better chice than a check for `null`. I'll update this pull request.

Please let me know if you want me to also include the additional `null` check before parsing the JSON.

Reply:
Thinking about this again, `angular.isObject` might hide some errors and it probably makes more sense to keep the `null` check. Still, let me know if you want another one before the `JSON.parse`.